patches/minecraft/net/minecraft/tags/TagContainer.java.patch

```diff
@@ -1,22 +1,21 @@
 --- a/net/minecraft/tags/TagContainer.java
 +++ b/net/minecraft/tags/TagContainer.java
-@@ -15,7 +_,7 @@
- import org.apache.logging.log4j.LogManager;
- import org.apache.logging.log4j.Logger;
- 
--public class TagContainer {
-+public class TagContainer implements net.minecraftforge.common.extensions.IForgeTagContainer {
-    static final Logger f_144432_ = LogManager.getLogger();
-    public static final TagContainer f_13420_ = new TagContainer(ImmutableMap.of());
-    public final Map<ResourceKey<? extends Registry<?>>, TagCollection<?>> f_144433_;
 @@ -74,6 +_,7 @@
     public void m_13431_() {
        StaticTags.m_13269_(this);
        Blocks.m_50758_();
-+      net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.TagsUpdatedEvent.VanillaTagTypes(this));
++      net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.TagsUpdatedEvent(this));
     }
  
     public Map<ResourceKey<? extends Registry<?>>, TagCollection.NetworkPayload> m_144442_(final RegistryAccess p_144443_) {
+@@ -81,6 +_,7 @@
+       this.m_144436_(new TagContainer.CollectionConsumer() {
+          public <T> void m_142574_(ResourceKey<? extends Registry<T>> p_144481_, TagCollection<T> p_144482_) {
+             Optional<? extends Registry<T>> optional = p_144443_.m_6632_(p_144481_);
++            optional = optional.isPresent() ? optional : net.minecraftforge.common.ForgeTagHandler.getWrapperRegistry(p_144481_);
```

Comment:
I feel like this may be slightly cleaner moving the optional.isPresent() -> optional return into the getWrapperRegistry method